src/udev/udevd.c

```diff
@@ -1151,6 +1153,9 @@ static int on_uevent(sd_device_monitor *monitor, sd_device *dev, void *userdata)
 
         (void) event_queue_assume_block_device_unlocked(manager, dev);
 
+        /* To make the stack directory /run/udev/links cleaned up later. */
+        manager->udev_node_needs_cleanup = true;
```

Comment:
why not just set this when actually creating/changing the dir?

Reply:
Creating/changing the directory is done by workers. But cleanup must be done by the manager on idle. Hence, the flag must be set by the main process.